src/main/kotlin/cacophony/grammar/syntaxtree/Expression.kt

```diff
@@ -0,0 +1,279 @@
+            range: Pair<Location, Location>,
+        ) : Expression(range)
+
+        class KeywordIf(
+            range: Pair<Location, Location>,
+        ) : Expression(range)
+
+        class KeywordThen(
+            range: Pair<Location, Location>,
+        ) : Expression(range)
+
+        class KeywordElse(
+            range: Pair<Location, Location>,
+        ) : Expression(range)
+
+        class KeywordWhile(
+            range: Pair<Location, Location>,
+        ) : Expression(range)
+
+        class KeywordDo(
+            range: Pair<Location, Location>,
+        ) : Expression(range)
+    }
+
+    // expression in parenthesis
+    class NestedExpression(
+        range: Pair<Location, Location>,
+        val expression: Expression,
+    ) : Expression(range)
+
+    // series of expressions separated by semicolons, returning value of the last one
+    // they can maybe be merged with NestedExpression, but I don't know if that would be convenient
+    // we agreed that (x; y;) returns Unit - should we create artificial class for empty expression after the last semicolon,
+    // or will we handle it in a different way?
+    class SubsequentExpressions(
+        range: Pair<Location, Location>,
+        vararg val expressions: Expression,
+    ) : Expression(range)
+
+    sealed class Statement(
+        range: Pair<Location, Location>,
+    ) : Expression(range) {
+        class IfStatement(
+            range: Pair<Location, Location>,
+            val keywordIf: Keyword.KeywordIf,
+            val testExpression: Expression,
+            val keywordThen: Keyword.KeywordThen,
+            val doExpression: Expression,
+        ) : Statement(range)
+
+        class IfElseStatement(
+            range: Pair<Location, Location>,
+            val keywordIf: Keyword.KeywordIf,
+            val testExpression: Expression,
+            val keywordThen: Keyword.KeywordThen,
+            val doExpression: Expression,
+            val keywordElse: Keyword.KeywordElse,
+            val elseExpression: Expression,
+        ) : Statement(range)
+
+        class WhileStatement(
+            range: Pair<Location, Location>,
+            val keywordWhile: Keyword.KeywordWhile,
+            val testExpression: Expression,
+            val keywordDo: Keyword.KeywordDo,
+            val doExpression: Expression,
+        ) : Statement(range)
+
+        class ReturnStatement(
+            range: Pair<Location, Location>,
+            val keywordReturn: Keyword.KeywordReturn,
+            val value: Expression,
+        ) : Statement(range)
+    }
+
+    class FunctionDefinition(
+        range: Pair<Location, Location>,
+        val functionName: VariableDeclaration,
+        vararg val arguments: FunctionArgument,
+        val returnType: Type,
+    ) : Expression(range)
+
+    class FunctionCall(
+        range: Pair<Location, Location>,
+        val functionName: Variable,
+        vararg val arguments: Variable,
+    ) : Expression(range)
```

Comment:
**issue:** I don't think this works for scenarios like
`f[1][2]`, which should parse to something like
```
FunctionCall(FunctionCall("f", 1), 2)
```
I think `functionName` should be any `Expression`